src/Fixer/ArrayNotation/AlphabeticalArrayKeySortFixer.php

```diff
@@ -0,0 +1,315 @@
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\ArrayNotation;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\Fixer\ConfigurableFixerInterface;
+use PhpCsFixer\FixerConfiguration\FixerConfigurationResolver;
+use PhpCsFixer\FixerConfiguration\FixerConfigurationResolverInterface;
+use PhpCsFixer\FixerConfiguration\FixerOptionBuilder;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\FixerDefinition\FixerDefinitionInterface;
+use PhpCsFixer\Preg;
+use PhpCsFixer\Tokenizer\CT;
+use PhpCsFixer\Tokenizer\Tokens;
+
+/**
+ * @author Abdurrahman Uymaz <abdurrahman.uymaz@mobian.global>
+ * @author Lars Grevelink <lars.grevelink@mobian.global>
+ * @author Leander Philippo <leander.philippo@mobian.global>
+ */
+final class AlphabeticalArrayKeySortFixer extends AbstractFixer implements ConfigurableFixerInterface
+{
+    public function getDefinition(): FixerDefinitionInterface
+    {
+        return new FixerDefinition(
+            'Sorts keyed arrays alphabetically.',
+            [
+                new CodeSample(
+                    "<?php\n\$sample = array('b' => '2', 'a' => '1', 'd' => '5');\n"
+                ),
+                new CodeSample(
+                    "<?php\n\$sample = ['b' => '2', 'a' => '1', 'd' => '5'];\n",
+                    []
+                ),
+                new CodeSample(
+                    "<?php\n\$sample = ['b' => '2', 'a' => '1', foo() => 'bar', 'd' => '5'];\n",
+                    ['special_keys_placement' => 'bottom']
+                ),
+                new CodeSample(
+                    "<?php\n\$sample = ['b' => '2', 'a' => '1', foo() => 'bar', 'd' => '5'];\n",
+                    ['special_keys_placement' => 'top']
+                ),
+            ],
+            'Use this fixer when you want a standardised representation of your keyed arrays; it will sort the numeric and string keys of your arrays alphabetically. Special keys, like concatenated ones, are kept in the same order but are either moved to the top or bottom (default) according to the configuration option.',
+            'Risky when the order of the array has an impact on the code execution.'
+        );
+    }
+
+    public function isRisky(): bool
+    {
+        return true;
+    }
+
+    public function isCandidate(Tokens $tokens): bool
+    {
+        return $tokens->isAnyTokenKindsFound([T_ARRAY, CT::T_ARRAY_SQUARE_BRACE_OPEN]);
+    }
+
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens): void
+    {
+        $this->sortTokens($tokens);
+    }
+
+    protected function createConfigurationDefinition(): FixerConfigurationResolverInterface
+    {
+        return new FixerConfigurationResolver([
+            (new FixerOptionBuilder('special_keys_placement', 'Whether to sort the specials keys on the bottom `bottom` or top `top`.'))
```

Comment:
what keys are "special" ?

Reply:
Keys that are expressions, like function calls, concatenated strings etc.